R/overlay.R

```diff
@@ -1,27 +1,24 @@
-## Sara Orofino
-## January 31, 2022
-## Spatial Overlap: Pipeline Step 01
-
-
 #' Function to find spatial overlap between a raster and a set of polygons
 #'
-#' @param data_source the source of climate data (default is era5)
-#' @param input_polygons a simple features polygon or multipolygon object
+#' @param polygons a simple features polygon or multipolygon object
 #' @param polygon_id the name of a column in the sf object representing a unique
 #'   identifier for each polygon
-#' @param weights_table an optional data table of secondary weights
+#' @param grid a raster layer with the same spatial resolution as the data
+#' @param secondary_weights an optional data table of secondary weights, output
```

Comment:
This is minor and I think my mistake since I wrote it initially, but I think we should be careful with use of the phrase data table; if this actually requires an input that is data.table format then we can say specify that otherwise maybe just use "optional secondary weights".

Reply:
I actually think that was me. As far as I know it does require the data.table produced by the previous step. This reference to data.table comes up quite often in the existing documentation so if it should be changed let me know and I'll go through it. I agree that at the very least it needs to be data.table and not data table

Reply:
I think if you use `data.table::as.data.table(secondary_weights)` (which I think I saw at some point) then it doesn't actually have to be in data.table format already because you're converting a data frame or tibble or whatever the input is to a data.table. And yea I think I was maybe liberal with using data table as a phrase instead of data frame or tibble or something else so for the parameter documentation at the top it's probably important that we double check so we aren't being misleading (sorry for making that harder for you!), and yes definitely to data.table over data table if that is required